File: win_scripts/wsh.jscript/h00.associative.js
// initialize Array object
var ages = {};

// add one elment at a time (using subscript)
ages["bob"]   = 34;
ages["ed"]    = 58;
ages["steve"] = 32;
ages["ralph"] = 23;
ages["deb"]   = 46;
ages["kate"]  = 19;

// enumerate & print values
WScript.stdout.write("Keys (names): ");
for (var key in ages) WScript.stdout.write(" " + key);
WScript.stdout.write("\n")
WScript.stdout.write("Values (ages):");
for (var key in ages) WScript.stdout.write(" " + ages[key]);
WScript.stdout.write("\n")

File: compiled_lang/go/src/h00.associative.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	// create empty map
	ages := make(map[string]int)
	// insert one element at a time
	ages["bob"] = 34
	ages["ed"] = 58
	ages["steve"] = 32
	ages["ralph"] = 23
	ages["deb"] = 46
	ages["kate"] = 19

	// enumerate and print keys and values
	keys := make([]string, 0, len(ages))
	values := make([]string, 0, len(ages))
	for k, v := range ages {
		keys = append(keys, k)
		values = append(values, fmt.Sprintf("%d", v))
	}
	fmt.Println("Keys (names):  " + strings.Join(keys, ", "))
	fmt.Println("Values (ages): " + strings.Join(values, ", "))
}
